conans/test/functional/toolchains/gnu/autotools/test_win_bash.py

```diff
@@ -0,0 +1,78 @@
+import platform
+import textwrap
+
+import pytest
+
+from conans.test.assets.autotools import gen_makefile_am, gen_configure_ac
+from conans.test.assets.sources import gen_function_cpp
+from conans.test.conftest import tools_locations
+from conans.test.functional.utils import check_exe_run
+from conans.test.utils.tools import TestClient
+from conans.util.files import save
+
+
+@pytest.mark.skipif(platform.system() not in ["Windows"], reason="Requires Windows")
+@pytest.mark.tool_msys2()
+def test_autotools_bash_complete():
+    client = TestClient(path_with_spaces=False)
+    bash_path = tools_locations["msys2"]["Windows"]["default"] + "/bash.exe"
+    save(client.cache.new_config_path, textwrap.dedent("""
+            tools.win.bash:subsystem=msys2
+            tools.win.bash:path={}
+            """.format(bash_path)))
+
+    main = gen_function_cpp(name="main")
+    # The autotools support for "cl" compiler (VS) is very limited, linking with deps doesn't
+    # work but building a simple app do
+    makefile_am = gen_makefile_am(main="main", main_srcs="main.cpp")
+    configure_ac = gen_configure_ac()
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        from conan.tools.gnu import Autotools
+        from conan.tools.microsoft.toolchain import write_conanvcvars
+        from conan.tools.env import save_script, Environment
+
+        class TestConan(ConanFile):
+            settings = "os", "compiler", "arch", "build_type"
+            exports_sources = "configure.ac", "Makefile.am", "main.cpp"
+            generators = "AutotoolsToolchain"
+            win_bash = True
+
+            def generate(self):
+                # Add vcvars launcher
```

Comment:
@memsharded Please take a look at these two FIXME in case we want to add more functionality now.

Reply:
If it is going to be exposed to public already, then maybe yes, add a generator like class. But can be done in a later PR.

Reply:
Ok. created https://github.com/conan-io/conan/issues/9213